src/odemis/acq/stream/_projection.py

```diff
@@ -218,6 +224,604 @@ def _updateImage(self):
+                polarimetry_cache[ebeam_pos] = {}
+
+                # TODO invalid division check if only bg  # norm so far, happens only when starting GUI newly
+
+                # select a color map based on the data
+                if pol_pos in [MD_POL_EPHI, MD_POL_ETHETA, MD_POL_EX, MD_POL_EY, MD_POL_EZ]:  # TODO: hardcoded for now
+                    data = numpy.abs(cache_raw[pol_pos])
+                    try:
+                        plotorder = int(numpy.log10(data.max()))
+                    except OverflowError:
+                        # Note: Error is raised, when bg image and image are the same
+                        # (-> cannot convert float infinity to integer)
+                        plotorder = 0
+
+                    cache_raw[pol_pos] = data / 10 ** plotorder
+                    colormap = _CMAP_EELS
+                else:
+                    colormap = _CMAP_MIT
+
+                # Convert the data to polar representation for GUI display.
+                polarimetry_data = angleres.Rectangular2Polar(cache_raw[pol_pos], colormap=colormap)
+
+                new_md = self._find_metadata(polarimetry_data.metadata)
+                new_md[model.MD_DIMS] = "YXC"
+                polarimetry_cache[ebeam_pos][pol_pos] = model.DataArray(polarimetry_data, new_md)
+
+                # return an array
+                polarimetry_data = polarimetry_cache[ebeam_pos][pol_pos]
+
+            except Exception:
+                logging.exception("Failed to convert the raw polarimetry data to RGB polar representation.")
+                return None
+
+        return polarimetry_data
+
+    def _updateImage(self):
+        """
+        Recomputes the image for the current ebeam position and polarimetry visualization requested.
+        """
+
+        if not self.raw:
+            return
+
+        # TODO check looks like it is called twice when loading a new image/bg image
+        # TODO most likely on the histogram update, calc histogram in this class
+
+        ebeam_pos = self.stream.point.value
+        try:
+            if ebeam_pos == (None, None):
+                self.image.value = None
+            else:
+                pol_pos = self.stream.polarimetry.value
+
+                # Calculate the raw polarimetry data (phi/theta representation).
+                # Note: If already done before for the requested pixel (ebeam pos), method will immediately return.
+                cache_raw = self._projectAsRaw(ebeam_pos)
+                # TODO might want to rename this method as we use project2Raw for csv export
+                if cache_raw is None:
+                    self.image.value = None
+                    return
+
+                # Project the raw polarimetry data to RGB polar representation.
+                polarimetry_data = self._project2RGBPolar(ebeam_pos, pol_pos, cache_raw)
+
+                # TODO histogram on projection instead of stream
```

Comment:
Yes, that would be a nice improvement. Unfortunately, that requires quite some changes also in the GUI, so that it shows the "right" histogram in the stream panel based on the current view (currently, it's just hard-coded to the stream histogram).